engineapi/backups.go

```diff
@@ -22,6 +23,16 @@ const (
 	backupStateError      = "error"
 )
 
+type BackupTargetBinaryClient interface {
```

Comment:
This is interface not implementation, so suggest call `BackupTargetClient`?

Reply:
Talked to @innobead , the name is taken as [struct](https://github.com/longhorn/longhorn-manager/blob/master/engineapi/backups.go#L25) already. Keep it for now.

Reply:
Thanks, @c3y1huang . 

note: I feel our naming pattern requires some improvement and consistency especially to differentiate the interface and structure but this can be tackled later. (this is common in all languages)